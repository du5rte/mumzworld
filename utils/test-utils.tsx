import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from '@shopify/restyle';
import { lightTheme } from '@/styles/themes';
import '@/locales/i18next-jest';

const AllTheProviders = ({ children }: PropsWithChildren) => {
  return <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>;
};

const customRender: typeof render = (ui, options?) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
