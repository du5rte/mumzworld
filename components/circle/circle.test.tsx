import { render, screen } from '@/utils/test-utils';

import Circle from './circle';

describe('circle', () => {
  it('should renders correctly', () => {
    render(<Circle size={100} backgroundColor="primary" />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
