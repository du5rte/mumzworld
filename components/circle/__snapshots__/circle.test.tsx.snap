// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle should renders correctly 1`] = `
<View
  size={100}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "hsl(0, 0%, 0%)",
        "borderRadius": 9999,
        "height": 100,
        "justifyContent": "center",
        "width": 100,
      },
    ]
  }
/>
`;
